src/pytorch_lightning/core/module.py

```diff
@@ -1950,6 +1951,70 @@ def _register_sharded_tensor_state_dict_hooks_if_available(self) -> None:
                 weakref.proxy(self), pre_load_state_dict_hook, True  # type: ignore[arg-type]
             )
 
+    @classmethod
+    def from_compiled(cls, model: Union["pl.LightningModule", "torch._dynamo.OptimizedModule"]) -> "pl.LightningModule":
```

Comment:
Typed as LightningModule or OptimizedModule but then raises an error for anything that isn't an OptimizedModule:
```suggestion
    def from_compiled(cls, model: "torch._dynamo.OptimizedModule") -> "pl.LightningModule":
```

Reply:
great catch!